projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts

```diff
@@ -124,3 +148,44 @@ describe('ReportBPComponent', () => {
   });
 
 });
+
+describe('snackbar test', () => {
+  let fixture: ComponentFixture<ReportBPComponent>;
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule, NoopAnimationsModule, MatInputModule, MatCardModule],
+      declarations: [ReportBPComponent],
+      providers: [ MatSnackBar],
+    }).compileComponents();
+    fixture = TestBed.createComponent(ReportBPComponent);
+    fixture.detectChanges();
+    loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+  });
+
+  it('should load harness for simple snack-bar', async () => {
+    const snackBarRef = fixture.componentInstance.open('Hi!', '');
+    let snackBars = await loader.getAllHarnesses(MatSnackBarHarness);
+
+    expect(snackBars.length).toBe(1);
+
+    snackBarRef.dismiss();
+    snackBars = await loader.getAllHarnesses(MatSnackBarHarness);
+    expect(snackBars.length).toBe(0);
+  });
+
+  it('should be able to get snackbar message after submit ', fakeAsync(async () => {
+    const systolicInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='systolic']" }));
+    await systolicInputHarness.setValue('100');
+    const diastolicInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='diastolic']" }));
+    await diastolicInputHarness.setValue('50');
+    const submitButtonHarness = await loader.getHarness(MatButtonHarness.with({ selector: "[id='submit']" }));
+    await submitButtonHarness.click();
+    tick(5000);
+    fixture.componentInstance.open('Blood Pressure Added Sucessfully')
+    let snackBar = await loader.getHarness(MatSnackBarHarness);
+    expect(await snackBar.getMessage()).toBe('Blood Pressure Added Sucessfully');
```

Comment:
The component's `open` method just calls `MatSnackBar.open` with the exact same parameters. That's not the part that we need to test. The complexity is in the `onSubmit` method where it is handling the promise resolution/rejection and deciding which parameters to pass in to the snackbar.

Reply:
@semicolin  I tried many times but still got the error element not found.
Can you please help me to fix this?

Reply:
yes, I will take a look

Reply:
@semicolin, i have to spy   `const snackBarSpy = spyOn(MatSnackBar.prototype, 'open').and.stub(); `
 like this used to prototype, The reason to use MatSnackBar.prototype in the spy is that MatSnackBar is not a constructor function. Instead, it is a class with a create method that returns an instance of MatSnackBar. In order to spy on the open method of MatSnackBar, we need to access it through the prototype object of the class.

Reply:
Can't we just use `createSpyObj` like we do with every other class that we need to spy on? 
`jasmine.createSpyObj<MatSnackBar>('MatSnackBar', ['open']);`